[Bug]: flang installation fails

### Problem description

Package `flang` fails to install, resulting in a partial install.

Without it, I don't have a FORTRAN compiler that works under Termux.

Find attached the installer output.
[flang.err.gz](https://github.com/termux/termux-packages/files/14417370/flang.err.gz)


### What steps will reproduce the bug?


pkg install flang
...
error: Semantic errors in /data/data/com.termux/files/usr/opt/flang/module/__ppc_intrinsics.f90
/data/data/com.termux/files/usr/opt/flang/module/__ppc_intrinsics.f90:39:3: error: Vector type is only supported for PowerPC
    ELEM_FUNC_VIVI(1) ELEM_FUNC_VIVI(2) ELEM_FUNC_VIVI(4) ELEM_FUNC_VIVI(8)
... 

### What is the expected behavior?

the package should install normally

### System information

termux-info:

```
Termux Variables:
TERMUX_API_VERSION=0.50.1
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=939
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
TERMUX__USER_ID=0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://packages.termux.dev/apt/termux-main stable main
# x11-repo (sources.list.d/x11.list)
deb https://packages.termux.dev/apt/termux-x11 x11 main
Updatable packages:
openjpeg/stable 2.5.1 aarch64 [upgradable from: 2.5.0-1]
termux-tools version:
1.40.6
Android version:
12
Kernel build information:
Linux localhost 4.19.191 #1 SMP PREEMPT Fri Aug 25 12:07:06 CST 2023 aarch64 Android
Device manufacturer:
jumper
Device model:
EZpad M10 SE
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
Installed termux plugins:
com.termux.api versionCode:51

```

--- a/packages/flang/build.sh
+++ b/packages/flang/build.sh
@@ -4,13 +4,14 @@ TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_LICENSE_FILE="flang/LICENSE.TXT"
 TERMUX_PKG_MAINTAINER="@termux"
 LLVM_MAJOR_VERSION=17
-TERMUX_PKG_VERSION=${LLVM_MAJOR_VERSION}.0.3
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_VERSION=${LLVM_MAJOR_VERSION}.0.6
 TERMUX_PKG_SRCURL=https://github.com/llvm/llvm-project/releases/download/llvmorg-$TERMUX_PKG_VERSION/llvm-project-$TERMUX_PKG_VERSION.src.tar.xz
-TERMUX_PKG_SHA256=be5a1e44d64f306bb44fce7d36e3b3993694e8e6122b2348608906283c176db8
+TERMUX_PKG_SHA256=58a8818c60e6627064f312dbf46c02d9949956558340938b71cf731ad8bc0813
 TERMUX_PKG_AUTO_UPDATE=false
 TERMUX_PKG_HOSTBUILD=true
-TERMUX_PKG_DEPENDS="libc++, libllvm, clang, lld, mlir"
+# `flang-new` should be rebuilt when libllvm bumps version.
+# See https://github.com/termux/termux-packages/issues/19362
+TERMUX_PKG_DEPENDS="libc++, libllvm (= $TERMUX_PKG_VERSION), clang (= $TERMUX_PKG_VERSION), lld (= $TERMUX_PKG_VERSION), mlir (= $TERMUX_PKG_VERSION)"
 TERMUX_PKG_BUILD_DEPENDS="libllvm-static"
 
 # Upstream doesn't support 32-bit arches well. See https://github.com/llvm/llvm-project/issues/57621.

--- a/packages/flang/postinst.sh.in
+++ b/packages/flang/postinst.sh.in
@@ -21,14 +21,14 @@ generate_module_files() {
     for file in ${_modules[@]}; do
         echo "Generating ${file}.mod..."
         cd "$TERMUX_PREFIX"/include/flang
-        flang-new -cpp -fsyntax-only $TERMUX_PREFIX/opt/flang/module/"$file".f90
+        flang-new -cpp -fsyntax-only -module-dir "$TERMUX_PREFIX"/include/flang "$TERMUX_PREFIX"/opt/flang/module/"$file".f90
         cp "$file".mod "$file".f18.mod
     done
 
     file="__ppc_intrinsics"
     echo "Generating ${file}.mod..."
     cd "$TERMUX_PREFIX"/include/flang
-    flang-new -cpp -fsyntax-only --target=ppc64le $TERMUX_PREFIX/opt/flang/module/"$file".f90
+    flang-new -cpp -fsyntax-only --target=ppc64le -module-dir "$TERMUX_PREFIX"/include/flang "$TERMUX_PREFIX"/opt/flang/module/"$file".f90
     cp "$file".mod "$file".f18.mod
 }
 

